internal/controller/runtime/fsm/runtime_fsm_create_shoot.go

```diff
@@ -58,3 +61,18 @@ func sFnCreateShoot(ctx context.Context, m *fsm, s *systemState) (stateFn, *ctrl
 
 	return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
 }
+
+func convertCreate(instance *imv1.Runtime, cfg config.ConverterConfig) (gardener.Shoot, error) {
+	if err := instance.ValidateRequiredLabels(); err != nil {
+		return gardener.Shoot{}, err
+	}
+
+	converter := gardener_shoot.NewConverterCreate(cfg)
+	newShoot, err := converter.ToShoot(*instance)
+	if err != nil {
+		setObjectFields(&newShoot)
```

Comment:
The call of `setObjectFields` is not needed here. The `setObjectFields` function was used only for patching existing shoot with some properties and used as workaroud for client limitations of missing those properties after exeturing " m.ShootClient.Get" operation in `sFnTakeSnapshot` function.
For "create mode" of converter setting object fields can be done in the converter context, or in `ToShoot` method directly.